flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -428,6 +450,9 @@ CheckpointTriggerResult triggerCheckpoint(
 			catch (Throwable t) {
 				int numUnsuccessful = numUnsuccessfulCheckpointsTriggers.incrementAndGet();
 				LOG.warn("Failed to trigger checkpoint (" + numUnsuccessful + " consecutive failed attempts so far)", t);
+				if(numUnsuccessful > maxUnsuccessfulCheckpoints) {
```

Comment:
You mean other than the existing 'numUnsuccessfulCheckpointsTriggers'? The reset is happening already in the code.